tests/app/views/test_brief_response.py

```diff
@@ -326,26 +389,140 @@ def test_day_rate_should_be_less_than_service_max_price(self, live_dos_framework
         assert data == {'error': {'dayRate': 'max_less_than_min'}}
 
 
-class TestSubmitBriefResponse(BaseBriefResponseTest):
+class TestUpdateBriefResponse(BaseBriefResponseTest):
+    def _create_brief_response(self):
+        res = self.client.post(
+            '/brief-responses',
+            data=json.dumps({
+                'updated_by': 'test@example.com',
+                'briefResponses': {
+                    'briefId': self.brief_id,
+                    'supplierId': 0
+                },
+                'page_questions': []
+            }),
+            content_type='application/json'
+        )
+        self.brief_response_id = json.loads(res.get_data(as_text=True))['briefResponses']['id']
 
-    def _setup_existing_brief_response(self, brief_response_data):
-        res = self.create_brief_response(dict(brief_response_data, **{
+    def test_brief_response_can_be_updated(self, live_dos_framework):
+        self._create_brief_response()
+
+        res = self._update_brief_response(self.brief_response_id, {'respondToEmailAddress': 'newemail@email.com'})
+        assert res.status_code == 200
+        data = json.loads(res.get_data(as_text=True))['briefResponses']
+        assert data['id'] == self.brief_response_id
+        assert data['briefId'] == self.brief_id
+        assert data['supplierId'] == 0
+        assert data['respondToEmailAddress'] == 'newemail@email.com'
+
+    def test_update_brief_response_creates_audit_event(self, live_dos_framework):
+        self._create_brief_response()
+
+        res = self._update_brief_response(self.brief_response_id, {'respondToEmailAddress': 'newemail@email.com'})
+        assert res.status_code == 200
+
+        with self.app.app_context():
+            audit_events = AuditEvent.query.filter(
+                AuditEvent.type == AuditTypes.update_brief_response.value
+            ).all()
+
+        assert len(audit_events) == 1
+        assert audit_events[0].data == {
+            'briefResponseId': self.brief_response_id,
+            'briefResponseData': {'respondToEmailAddress': 'newemail@email.com'}
+        }
+
+    def test_update_brief_response_that_does_not_exist_will_404(self, live_dos_framework):
+        res = self._update_brief_response(100, {'respondToEmailAddress': 'newemail@email.com'})
+        assert res.status_code == 404
+
+    def test_can_not_update_brief_response_for_framework_that_is_not_live(self, live_dos_framework):
+        self._create_brief_response()
+
+        with self.app.app_context():
+            # Change live framework to be expired
+            db.session.execute("UPDATE frameworks SET status='expired' WHERE slug='digital-outcomes-and-specialists'")
```

Comment:
After some digging, it looks like it does. 

Because this test class uses the `live_dos_framework` fixture, the next test that is run after this will make sure that there is a live dos framework available in the database. 

The way this is done is quite smart, that because dos already exists in the database, it will update the dos record in the database as opposed to creating a new one. Thus setting dos back to 'live'. (see https://github.com/alphagov/digitalmarketplace-api/blob/master/tests/conftest.py#L114).

This isn't perfect however because this behaviour only happens because subsequent tests use the `live_dos_framework` fixture. If this was the last test that used that fixture and then the next tests were expected dos to be live then I'm not sure they would pass. This isn't the case, so this currently won't cause any issues. Do we think it's worth doing to be extra safe?